### PR TITLE
Add adoption task to nova role

### PR DESCRIPTION
**Pull request by mrkisaolamb**

### CODE REVIEW

File: roles/edpm_nova/tasks/adoption.yml [hunk 1]
```diff
@@ -0,0 +1,16 @@
+---
+- name: Stop and disable compute services
+  no_log: "{{ edpm_network_config_hide_sensitive_logs | bool }}"
+  tags:
+    - adoption
+    - nova
+  become: true
+  ansible.builtin.shell: |
```
Comment (gibizer) [inline roles/edpm_nova/tasks/adoption.yml:8]: can we use ansible.builtin.systemd instead?

File: roles/edpm_nova/defaults/main.yml [hunk 2]
```diff
@@ -31,3 +31,14 @@ edpm_nova_compute_image: "quay.io/podified-antelope-centos9/openstack-nova-compu
 
 # certs
 edpm_nova_tls_certs_enabled: "{{ edpm_tls_certs_enabled | default(False) }}"
+# default list example
+edpm_nova_compute_services_to_stop:
```
Comment (gibizer) [inline roles/edpm_nova/defaults/main.yml:35]: would use a name that signals that these are old tripleo services. What about `edpm_old_tripleo_compute_sevices`

File: roles/edpm_nova/defaults/main.yml [hunk 3]
```diff
@@ -31,3 +31,14 @@ edpm_nova_compute_image: "quay.io/podified-antelope-centos9/openstack-nova-compu
 
 # certs
 edpm_nova_tls_certs_enabled: "{{ edpm_tls_certs_enabled | default(False) }}"
+# default list example
```
Comment (gibizer) [inline roles/edpm_nova/defaults/main.yml:34]: probably copy paste

File: roles/edpm_nova/tasks/adoption.yml [hunk 4]
```diff
@@ -0,0 +1,16 @@
+---
+- name: Stop and disable compute services
+  no_log: "{{ edpm_network_config_hide_sensitive_logs | bool }}"
```
Comment (gibizer) [inline roles/edpm_nova/tasks/adoption.yml:3]: hm, this variable seems to be intended to the network_config roles. Do we even foresee some sensitive logs here?

**Review by gibizer — COMMENTED** [review]
Do we want to disable these adoption tasks in the greenfield deployment? If so, we need to somehow disable the adoption tag in greenfield.

**Comment by gibizer** [comment]
I'm wondering how easy would be to create a meaningful molecule test for this task. @SeanMooney do you have some ideas how to simulate the existence of an old tripleo service in molecule?

**Comment by gibizer** [comment]
@mrkisaolamb I'm not sure depends-on works between edpm-ansible and data-plane-adoption repo, but it worth a try to see if we can run this in the adoption CI even before merge

File: roles/edpm_nova/defaults/main.yml (hunk 3 above)
Comment (mrkisaolamb) [reply roles/edpm_nova/defaults/main.yml]: Done

File: roles/edpm_nova/tasks/adoption.yml (hunk 4 above)
Comment (mrkisaolamb) [reply roles/edpm_nova/tasks/adoption.yml]: done, removed

File: roles/edpm_nova/tasks/adoption.yml (hunk 1 above)
Comment (mrkisaolamb) [reply roles/edpm_nova/tasks/adoption.yml]: done

File: roles/edpm_nova/tasks/adoption.yml [hunk 5]
```diff
@@ -0,0 +1,12 @@
+---
+- name: Stop and disable compute services
+  tags:
+    - adoption
+    - nova
+  become: true
+  ansible.builtin.systemd:
+        name: "{{ item }}"
+        state: stopped
+        enabled: false
```
Comment (gibizer) [inline roles/edpm_nova/tasks/adoption.yml:10]: we mask them as well with `masked: yes`

**Review by gibizer — COMMENTED** [review]
Thanks. I think we need some pidfile deletion as well. Or have you checked, stopping the service removes the pidfiles too?

File: roles/edpm_nova/tasks/adoption.yml (hunk 5 above)
Comment (mrkisaolamb) [reply roles/edpm_nova/tasks/adoption.yml]: done

**Comment by mrkisaolamb** [comment]
recheck

File: roles/edpm_nova/tasks/adoption.yml (hunk 1 above)
Comment (bogdando) [reply roles/edpm_nova/tasks/adoption.yml:23]: have you pushed the updated revision yet?

Comment (mrkisaolamb) [reply roles/edpm_nova/tasks/adoption.yml:23]: I re-add this shell to simplified checking if service exist from list, task with systemd was failing if service wasn't running

**Comment by mrkisaolamb** [comment]
recheck

File: roles/edpm_nova/tasks/adoption.yml [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+  tags:
+    - adoption
+    - nova
+  block:
+    - name: Check if nova_virtlogd container exists
+      ansible.builtin.command: podman ps -a --filter name=nova_virtlogd --format \{\{.Names\}\}
+      register: nova_virtlogd_status
+      changed_when: false
+    - name: Stop nova_virtlogd containers not managed by service units
+      ansible.builtin.shell: "podman stop nova_virtlogd '||' true"
```
Comment (gibizer) [inline roles/edpm_nova/tasks/adoption.yml:14]: Does podman return a nonzero exitcode if it succeeds? Do we need to ignore all the exit codes?

**Comment by gibizer** [comment]
> Thanks. I think we need some pidfile deletion as well. Or have you checked, stopping the service removes the pidfiles too?

Did we settled this?

**Review by gibizer — APPROVED** [review]
So Kamil checked and the pidfiles are removed by the systemd service files. 
Also the systemd module is not used as based on Kamil it cannot stop already disabled services. 
The rest of my comments can be fixed in a follow up. So I approve this to see if it helps the adoption testing where we see virtlogd issues.

**Pull request merged**